sort/bucketsort.go

```diff
@@ -0,0 +1,46 @@
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// sorting an array using buckets
+// mainly useful when input is uniformly distributed over a range
+func BucketSort[T constraints.Number](arr []T) []T {
+	// early return if the array too small
+	if len(arr) <= 1 {
+		return arr
+	}
+
+	// find the maximum and minimum elements in arr
+	max := arr[0]
+	min := arr[0]
+	for _, v := range arr {
+		if v > max {
+			max = v
+		}
+		if v < min {
+			min = v
+		}
+	}
+
+	// create an empty bucket for each element in arr
+	bucket := make([][]T, len(arr))
+
+	// put each element in the appropriate bucket
+	for _, v := range arr {
+		bucketIndex := int((v - min) / (max - min) * T(len(arr)-1))
```

Comment:
I'm assuming this is the reason for the `constraint.Number` rather than the `constraints.Ordered`...

Reply:
yeah, exactly